tests/Tokenizer/Analyzer/Analysis/TypeAnalysisTest.php

```diff
@@ -123,75 +126,120 @@ public static function provideIsNullableCases(): iterable
         yield [false, '\foo\bar'];
 
         yield [true, '?\foo\bar'];
+    }
 
-        if (\PHP_VERSION_ID >= 8_00_00) {
-            yield [false, 'string|int'];
+    /**
+     * @dataProvider provideIsNullable80Cases
+     *
+     * @requires PHP 8.0
+     */
+    public function testIsNullable80(bool $expected, string $input): void
+    {
+        $analysis = new TypeAnalysis($input, 1, 2);
+        self::assertSame($expected, $analysis->isNullable());
+    }
 
-            yield [true, 'string|null'];
+    /**
+     * @return iterable<array{bool, string}>
+     */
+    public static function provideIsNullable80Cases(): iterable
+    {
+        yield [false, 'string|int'];
 
-            yield [true, 'null|string'];
+        yield [true, 'string|null'];
 
-            yield [true, 'string|NULL'];
+        yield [true, 'null|string'];
 
-            yield [true, 'NULL|string'];
+        yield [true, 'string|NULL'];
 
-            yield [true, 'string|int|null'];
+        yield [true, 'NULL|string'];
 
-            yield [true, 'null|string|int'];
+        yield [true, 'string|int|null'];
 
-            yield [true, 'string|null|int'];
+        yield [true, 'null|string|int'];
 
-            yield [true, 'string|int|NULL'];
+        yield [true, 'string|null|int'];
 
-            yield [true, 'NULL|string|int'];
+        yield [true, 'string|int|NULL'];
 
-            yield [true, 'string|NULL|int'];
+        yield [true, 'NULL|string|int'];
 
-            yield [false, 'string|\foo\bar'];
+        yield [true, 'string|NULL|int'];
 
-            yield [true, 'string|\foo\bar|null'];
+        yield [false, 'string|\foo\bar'];
 
-            yield [true, 'null|string|\foo\bar'];
+        yield [true, 'string|\foo\bar|null'];
 
-            yield [true, 'string|null|\foo\bar'];
+        yield [true, 'null|string|\foo\bar'];
 
-            yield [true, 'string |null| int'];
+        yield [true, 'string|null|\foo\bar'];
 
-            yield [true, 'string| null |int'];
+        yield [true, 'string |null| int'];
 
-            yield [true, 'string | null | int'];
+        yield [true, 'string| null |int'];
 
-            yield [false, 'Null2|int'];
+        yield [true, 'string | null | int'];
 
-            yield [false, 'string|Null2'];
+        yield [false, 'Null2|int'];
 
-            yield [false, 'string |Null2'];
+        yield [false, 'string|Null2'];
 
-            yield [false, 'Null2| int'];
+        yield [false, 'string |Null2'];
 
-            yield [false, 'string | Null2 | int'];
-        }
+        yield [false, 'Null2| int'];
 
-        if (\PHP_VERSION_ID >= 8_01_00) {
```

Comment:
good to have this moved to standalone PR. let's take care of that standalone PR first

Reply:
This was not moved anywhere - this PR adds the check and fixes all non-fixer tests classes.

Reply:
I thought you started moving those things in #7581 ?